packages/logs/src/domain/trackNetworkError.ts

```diff
@@ -46,6 +52,52 @@ export function trackNetworkError(configuration: LogsConfiguration, errorObserva
   }
 }
 
+// TODO: ideally, computeResponseData should always call the `callback` with a string instead of
+// `any`. But to keep retrocompatibility, in the case of XHR with a `responseType` different than
+// "text", the response data should be whatever `xhr.response` is. This is a bit confusing as Logs
+// event 'stack' is expected to be a string. This should be changed in a future major version as it
+// could be a breaking change.
+export function computeResponseData(
+  { xhr, response, error }: { xhr?: XMLHttpRequest; response?: Response; error?: unknown },
+  configuration: LogsConfiguration,
+  callback: (responseData?: any) => void
+) {
+  if (xhr) {
+    if (typeof xhr.response === 'string') {
+      callback(truncateResponseText(xhr.response, configuration))
+    } else {
+      callback(xhr.response)
+    }
+  } else if (response) {
+    if (!response.body) {
+      callback()
+    } else {
+      readBytes(
+        response.clone().body!,
+        // Read one more byte than the limit, so we can check if more bytes would be available and
+        // show an ellipsis in this case
+        configuration.requestErrorResponseLengthLimit + 1,
+        (error, bytes) => {
+          if (error) {
+            callback(`Unable to retrieve response: ${(error as unknown) as string}`)
+          } else {
+            let responseText = new TextDecoder().decode(bytes!.slice(0, configuration.requestErrorResponseLengthLimit))
+            if (bytes!.length > configuration.requestErrorResponseLengthLimit) {
+              responseText += '...'
+            }
+            callback(responseText)
+          }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏it could be nice to encapsulate all those concerns, maybe something like:

```suggestion
      truncateResponseStream(
        response.clone().body!,
        configuration.requestErrorResponseLengthLimit,
        (error, responseText) => {
          if (error) {
            callback(`Unable to retrieve response: ${(error as unknown) as string}`)
          } else {
            callback(responseText)
          }
        )
```